### DISCUSSION TITLE
Questions about monitoring

### DISCUSSION THREAD

**Discussion by xunleii** [goauthentik/authentik#6315] [Q&A] [answered]
Hello,

First of all, thank you very much for Authentik, it's a great project.
We're in the process of setting it up at Radio France and we have a few questions about monitoring (metrics to be more precise):
- What exactly do the `authentik_main_requests`, `authentik_outpost_proxy_requests` and `authentik_outpost_ldap_requests` metrics correspond to? Why use histograms? Their descriptions don't seem to match their names, and their values seem to be more response time than anything else (https://github.com/goauthentik/authentik/blob/main/internal/outpost/proxyv2/handlers.go#L23C16-L23C16).
- Are other metrics such as the number of people connected planned?

Thank you and have a nice day

**Comment by BeryJu** [comment]
`authentik_main_requests` measures the internal routing of the server container, which requests go to the backend server and which go to the embedded outpost. 
`authentik_outpost_proxy_requests` measures the requests going to an outpost (to which provider they go, etc)
`authentik_outpost_ldap_requests` is the number of LDAP requests LDAP types by provider and type (bind, unbind, search)

They are all histograms as they track the duration of each request 

I'm not quite sure what you mean with `the number of people connected`, there's a very wide definition of what connected can mean

**Reply by xunleii** [reply, marked as answer]
Thank you very much for your answer.

It's true about `the number of people connected`, it doesn't mean anything 😅. 
When I said that, I was thinking of the information present on the overview dashboard in the admin panel (like `Logins and permissions over the last week` or `Most used apps`). Also, other information like the number of active sessions could be really interesting as metrics that can be used inside a dashboard.

On the subject of histograms, I have two suggestions for improving them:
- Renaming these metrics following best practice by suffixing them with `_duration_seconds` (https://prometheus.io/docs/practices/naming/#metric-names) and changing their descriptions.
- Dividing all value on each `.Observe` call by `time.Seconds` to be consistent with the first suggestion (https://prometheus.io/docs/practices/naming/#base-units). Currently, the histograms are unusable because the values stored are in nanoseconds and the default buckets are designed to measure the response time in seconds (https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables).

I know that renaming these measures is a breaking change, but I don't think they really make sense in their current state.

If you find these suggestions useful, I can make a PR to make these changes if you wish.
